com.unity.shadergraph/Editor/Data/Implementation/NodeUtils.cs

```diff
@@ -132,11 +133,11 @@ public static SlotReference DepthFirstCollectRedirectNodeFromNode(RedirectNodeDa
                 {
                     var outputNode = edge.outputSlot.node;
                     if (outputNode != null)
-                        DepthFirstCollectNodesFromNode(nodeList, outputNode, keywordPermutation: keywordPermutation);
+                        DepthFirstCollectNodesFromNode(nodeList, outputNode, keywordPermutation: keywordPermutation, ignoreActiveState: ignoreActiveState);
                 }
             }
 
-            if (includeSelf == IncludeSelf.Include && node.isActive)
+            if (includeSelf == IncludeSelf.Include && (node.isActive || ignoreActiveState))
```

Comment:
This adds an option to the function to always gather the nodes, even if they are not flagged active.